inc/Editor/DataBinding/BlockBindings.php

```diff
@@ -204,18 +204,24 @@ private static function execute_queries( array $block_context, array $source_arg
 		}
 	}
 
-	public static function should_render_empty_result( WP_Block $block ): bool {
-		$block_context = $block->context[ self::$context_name ] ?? [];
+	public static function is_error_or_empty_state( array $context, array $attributes ): bool {
+		$block_context = $context[ self::$context_name ] ?? [];
 		// Re-execute the query to get the latest results, rather than using the
 		// stale results from the block.
 		$query_response = self::execute_queries( $block_context, [] );
 
+		// If there is an error, and it's the error block variation, return true.
 		if ( is_wp_error( $query_response ) ) {
-			return false;
+			return 'error' === $attributes['mode'] ?? 'unknown';
```

Comment:
My mistake in the suggestion. Order of operations matters a lot with the nullish coalescing operator. Either we should group or pass it just the `mode` attribute
```suggestion
			return 'error' === ( $attributes['mode'] ?? 'unknown' );
```

Reply:
I had originally just done the mode attribute and avoided the unknown so I'll go back to that.